bencode/src/com/github/lavrov/bencode/BencodeCodec.scala

```diff
@@ -1,19 +1,87 @@
 package com.github.lavrov.bencode
 
-import scodec.Codec
-import scodec.codecs._
-import scodec.Attempt
-import scodec.Err
-import scodec.SizeBound
-import scodec.DecodeResult
+import scodec.Attempt.{Failure, Successful}
 import scodec.bits.BitVector
-import scodec.Encoder
-import scodec.Decoder
-import scodec.bits.ByteVector
+import scodec.codecs._
+import scodec.{Attempt, Codec, DecodeResult, Decoder, Encoder, Err, SizeBound}
+
+import scala.util.Try
+
 
 object BencodeCodec {
-  val instance: Codec[Bencode] = {
-    lazily(???) // Implement codec!
+  val instance: Codec[Bencode] = lazily {
+    choice(bencodeString.upcast, bencodeInteger.upcast, bencodeList.upcast, bencodeDict.upcast)
+  }
+  
+  object AsciiNumber {
+    
+    private val positiveAsciiDigit: Codec[Short] = {
+      def decode(b: Byte): Attempt[Short] = {
+        Try(Character.toString(b.toChar).toShort) match {
+          case scala.util.Success(value) => Successful(value)
+          case scala.util.Failure(_) => Failure(Err("Can not decode"))
+        }
+      }
+      def encode(c: Short) =
+        Successful(c.toByte)
+      
+      byte.exmap(decode, encode)
+    }
+    
+    val positiveAsciiNumber: Codec[Long] = {
+      def listToLong(ds: List[Short]): Long = ds.reverse.zipWithIndex.foldLeft(0l){
```

Comment:
Doesn't cover the case when the list is empty. Probably this case is not covered by tests either ;)